R/tmb.R

```diff
@@ -472,6 +472,8 @@ fit_mmrm <- function(formula,
   covariance <- reconcile_cov_struct(formula, covariance)
   formula_parts <- h_mmrm_tmb_formula_parts(formula, covariance)
 
+  if (!formula_parts$is_spatial && !is.factor(data[[formula_parts$visit_var]])) stop("Time variable must be a factor for non-spatial covariance structures")
```

Comment:
```suggestion
  if (!formula_parts$is_spatial && !is.factor(data[[formula_parts$visit_var]])) {
    stop("Time variable must be a factor for non-spatial covariance structures")
  }
```

Reply:
I wonder though, now that we have the `validate_cov_struct()` function, would this assertion better fit there? can you please check?

Reply:
I don't think the `validate_cov_struct()` as is helps us here because the object is valid by construction. We could: 

1. Write a new `checkmate` style check, e.g., `validate_time_as_factor()`,
2. Include a new check within the `validate_cov_struct()` definition, and then run it here.

Reply:
Hm ok, then let's keep it as is. Thanks for checking!